File: Data Structures & Algorithms/minimum-window-with-characters/submission-0.cpp
class Solution {
public:
    string minWindow(string s, string t) {
        vector<int>hashs(60,0);
        vector<int>hasht(60,0);
        int left = 0;
        int ans = INT_MAX;
        pair<int,int> final_points;
        for(auto item:t) hasht[item-'A']++;
        int n = s.size();
        
        for(int right=0;right<n;right++){
            bool isvalid = 1;
            hashs[s[right]-'A']++;
            for(int i=0;i<60;i++){
                if(hashs[i]<hasht[i]){
                    isvalid=0;
                    break;         
                }
               
            }
            
             if(isvalid){
                cout<<left<<' '<<right<<endl;
            while(hashs[s[left]-'A']-1>=hasht[s[left]-'A']){
                hashs[s[left]-'A']--;
                left++;
            }
            if(right-left+1<ans){
                ans=right-left+1;
                final_points={left,right};
            }
        }
        }
       string ss="";
       cout<<ans<<endl;
       cout<<final_points.first<<' ';
        for(int i=final_points.first;i<=final_points.second;i++){
            ss.push_back(s[i]);
        }
        return ans==INT_MAX?"":ss;
    }
};

File: Data Structures & Algorithms/palindromic-substrings/submission-0.java
class Solution {
    public int countSubstrings(String s) {
        int n = s.length();
       boolean[][] dp=new boolean[n][n];
       int ans = 0;
       for(int l=1;l<=n;l++){
        for(int i=0;i<n-l+1;i++){
            int j=i+l-1;
            if(l==1) dp[i][j]=true;
            if(l==2 && s.charAt(i)==s.charAt(j)) dp[i][j]=true;
            if(l>2 && s.charAt(i)==s.charAt(j) && dp[i+1][j-1]==true) dp[i][j]=true;
            if(dp[i][j]==true)  ans++;
        }
       }

    return ans;
    }
}
